zboxcore/zboxutil/http.go

```diff
@@ -534,38 +535,35 @@ func MakeSCRestAPICall(scAddress string, relativePath string, params map[string]
 	entityResult := make(map[string][]byte)
 	var retObj []byte
 	maxCount := 0
+	wg := sync.WaitGroup{}
 	for _, sharder := range util.Shuffle(sharders) {
-		urlString := fmt.Sprintf("%v/%v%v%v", sharder, SC_REST_API_URL, scAddress, relativePath)
-		urlObj, _ := url.Parse(urlString)
-		q := urlObj.Query()
-		for k, v := range params {
-			q.Add(k, v)
-		}
-		urlObj.RawQuery = q.Encode()
-		client := &http.Client{Transport: DefaultTransport}
-
-		response, err := client.Get(urlObj.String())
-		if err != nil {
-			continue
-		} else {
-			defer response.Body.Close()
-			entityBytes, err := ioutil.ReadAll(response.Body)
-			if err != nil {
-				continue
+		wg.Add(1)
+		go func(sharder string) {
+			defer wg.Done()
+			urlString := fmt.Sprintf("%v/%v%v%v", sharder, SC_REST_API_URL, scAddress, relativePath)
+			urlObj, _ := url.Parse(urlString)
+			q := urlObj.Query()
+			for k, v := range params {
+				q.Add(k, v)
 			}
-			responses[response.StatusCode]++
-			if responses[response.StatusCode] > maxCount {
-				maxCount = responses[response.StatusCode]
-				retObj = entityBytes
+			urlObj.RawQuery = q.Encode()
+			client := &http.Client{Transport: DefaultTransport}
+
+			response, err := client.Get(urlObj.String())
+			if err == nil {
+				defer response.Body.Close()
+				entityBytes, _ := ioutil.ReadAll(response.Body)
+
+				responses[response.StatusCode]++
```

Comment:
The `responses` slice is not thread safe

Reply:
Added mutex locks to prevent race conditions.